llvm/lib/IR/Verifier.cpp

```diff
@@ -4255,9 +4255,10 @@ void Verifier::visitLoadInst(LoadInst &LI) {
     Check(LI.getOrdering() != AtomicOrdering::Release &&
               LI.getOrdering() != AtomicOrdering::AcquireRelease,
           "Load cannot have Release ordering", &LI);
-    Check(ElTy->isIntOrPtrTy() || ElTy->isFloatingPointTy(),
-          "atomic load operand must have integer, pointer, or floating point "
-          "type!",
+    Check(ElTy->getScalarType()->isIntOrPtrTy() ||
```

Comment:
This PR shouldn't touch the verifier, only handle codegen for atomic load / store

Reply:
It is stacked on top of https://github.com/llvm/llvm-project/pull/117625. Please take a look. This PR handles vectors only -- renamed the title.